stdlib/public/core/StringStorage.swift

```diff
@@ -298,6 +298,12 @@ final internal class __StringStorage
   final internal func isEqual(to other: AnyObject?) -> Int8 {
     return _isEqual(other)
   }
+  
+  @objc(isEqual:)
+  @_effects(readonly)
+  final internal func isEqual(object other:AnyObject?) -> Int8 {
+    return _isEqual(other)
```

Comment:
Doesn't `_isEqual` assume the other object is an NSString already?

Reply:
Nope. There's a comment in there about how we could do that according to the API contract, but are matching accidental behavior of NSString and not doing it.